kernel/init.c

```diff
@@ -40,6 +40,51 @@ LOG_MODULE_REGISTER(os, CONFIG_KERNEL_LOG_LEVEL);
 /* the only struct z_kernel instance */
 struct z_kernel _kernel;
 
+uint32_t init_multithread_enabled_mask_3_cnt = 0;
+uint32_t init_bg_thread_main_mask_3_cnt = 0;
+
+extern uint32_t my_int_mask_3_cnt;
+
+/* all 3 sections are cacheable */
+volatile uint32_t __aligned(64) __attribute__((section(".noinit"))) sof_status1[3] = {0x0, 0x0, 0x0};
+volatile uint32_t __aligned(64) __attribute__((section(".data"))) sof_status2[3] = {0x22221111, 0x22222222, 0x22223333};
+volatile uint32_t __aligned(64) __attribute__((section(".heap_mem"))) sof_status3[3] = {0x33331111, 0x33332222, 0x33333333};
+
+#ifdef CONFIG_IMX
```

Comment:
Obvoiusly none of this code belongs here.  If you want SOC-specific initialization hooks, they should be called out of the arch/soc layer before entry to z_cstart(), in arch_kernel_init() (called just after entry to z_cstart()), or via a standard device struct initialized via SYS_INIT

Reply:
This code is just for debugging - see commit https://github.com/zephyrproject-rtos/zephyr/pull/35467/commits/ea08a79fb2974dad505e9b7274b95bbd5ca2cd62

I've added several counters and status variables and I read them, from Linux, with /dev/mem.